test/complexrestriction.cc

```diff
@@ -39,13 +39,13 @@ void TestWriteRead() {
     throw runtime_error("ComplexRestriction to GraphId get failed");
   }
 
-  res.set_via_count(5);
+  res.set_via_list(std::vector<GraphId>(5, GraphId{}));
   if (res.via_count() != 5) {
     throw runtime_error("ComplexRestriction via count failed");
   }
 
-  res.set_via_count(kMaxViasPerRestriction + 5);
-  if (res.via_count() != kMaxViasPerRestriction) {
+  res.set_via_list(std::vector<GraphId>(kMaxViasPerRestriction + 7, GraphId{}));
+  if (res.via_count() != 5) {
```

Comment:
@gknisely I modified the test result to check for an _unchanged_ `via_count()`.

The previous code would set `via_count()` to `kMaxViasPerRestriction` when in-fact no via's were set at all due to the branch at https://github.com/valhalla/valhalla/pull/2152/files#diff-80d7f6da13d96be19ae752a925463c07R15